src/game.h

```diff
@@ -572,6 +572,10 @@ class game
         int assign_npc_id();
         Creature *is_hostile_nearby();
         Creature *is_hostile_very_close();
+        bool is_map_cache_valid = false;
```

Comment:
This probably doesn't belong to the `game` class. It's the `map`'s cache validity this flag denotes. It also should be private otherwise it defeats the purpose of `invalidate_map_cache()`.

Reply:
This is important, this MUST be owned by the map.

Reply:
I've looked into `map` class more thoroughly and found out it already had most of necessary functionality (i.e. it is possible to mark certain caches as dirty), so I utilized this to make new function which invalidates all three caches (transparency, floor and outside). Later we need to revisit places where we invalidate cache and only mark certain caches invalid (depending on context), but I think performance improvement should be noticeable already.